### PR TITLE
upgrade: document upgrade to v4.0

### PR DESCRIPTION
**Pull request by jrcastro2**
* closes https://github.com/inveniosoftware/invenio-app-rdm/issues/854

### CODE REVIEW

File: docs/releases/upgrading/upgrade-v4.0.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+The steps listed in this article require an existing local installation of InvenioRDM 3.0, please make sure that this is given!
+If unsure, run `invenio-cli install` from inside the instance directory before executing the listed steps.
+
+**Note**: Do *not* delete the old Python virtual environment, or the database migration may complain about missing packages.
+
+!!! warning "Backup"
+
+    Always backup your database and files before you try to perform an upgrade.
+
+!!! warning "Upgrade from v1 to v2"
```
Comment (lnielsen) [inline docs/releases/upgrading/upgrade-v4.0.md:15]: You can remove this one warning.

File: docs/releases/upgrading/upgrade-v4.0.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+This can be achieved by the following Bash shell commands:
+
+~~~bash
+# NOTE: make sure you're in the instance directory
+
+# Delete ES indices
+pipenv run invenio index destroy --yes-i-know
+
+# Upgrade packages
+rm Pipfile.lock
```
Comment (lnielsen) [inline docs/releases/upgrading/upgrade-v4.0.md:34]: invenio-cli packages update 4.0

File: docs/releases/upgrading/upgrade-v4.0.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+First, the Elasticsearch indices are deleted and the packages are upgraded. Then, the upgrade command will be executed, this will migrate the database, run the custom migration script and rebuild the Elasticsearch indices .
+
+This can be achieved by the following Bash shell commands:
+
+~~~bash
+# NOTE: make sure you're in the instance directory
+
+# Delete ES indices
+pipenv run invenio index destroy --yes-i-know
```
Comment (lnielsen) [inline docs/releases/upgrading/upgrade-v4.0.md:31]: Is this needed before?

Woudl be best if we could avoid it or at least make it before upgrade and after the package update

Comment (jrcastro2) [reply docs/releases/upgrading/upgrade-v4.0.md:31]: It's needed to delete the indexes `rdmrecords-drafts-draft-v1.0.0` and  `rdmrecords-records-record-v1.0.0`. I guess we can move this to the migration and we can remove it from the beginning. I will have a look tomorrow.

**Review by ppanero — APPROVED** [review]
LGTM, missing removal of `invenio` from the Pipfile. Added a card in the board to do so.

**Pull request merged**